docs/src/locators.md

```diff
@@ -367,6 +558,200 @@ await rowLocator
     .ScreenshotAsync();
 ```
 
+### Locate elements in Shadow DOM
+
+All locators in Playwright **by default** work with elements in Shadow DOM. The exceptions are:
+- Locating by XPath selector does not pierce shadow roots.
+- [Closed-mode shadow roots](https://developer.mozilla.org/en-US/docs/Web/API/Element/attachShadow#parameters) are not supported.
+
+Consider the following example with a custom web component:
+```html
+<x-badge>
+  <span>Title</span>
+  #shadow-root
+    <span>Details</span>
+</x-badge>
+```
+
+You can locate in the same way as if the shadow root was not present at all:
+
+```js
+// Clicks <span>Details</span>
```

Comment:
add text explanations outside of code blocks so we only have to maintain once. can be a list here or a paragraph